src/homework4.py

```diff
@@ -0,0 +1,421 @@
+            str(dif_w_mean_of_resp)
+        )
+    )
+    print(
+        "Difference with mean of response (weighted) = {}".format(
+            str(sum(weighted_mean_squared_diff))
+        )
+    )
+
+
+def random_forest(df, cont_list, response):
+    cont_list.append(response)
+    pd.set_option("mode.use_inf_as_na", True)
+
+    full_list = df[cont_list]
+    full_list = full_list.dropna()
+
+    X = full_list[cont_list]
+    X = X.drop(response, axis=1)
+    y = full_list[response]
+    cont_list = cont_list.remove(response)
+
+    classifier = RandomForestClassifier()
+    classifier.fit(X, y)
+
+    feature_imp = pd.Series(
+        classifier.feature_importances_, index=cont_list
+    ).sort_values(ascending=False)
+    print(feature_imp)
+
+
+def main():
+    # Given a pandas dataframe
+    # Contains both a response and predictors
+    df = pd.read_csv(
+        "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/titanic.csv"
+    )
+
+    # Given a list of predictors and the response columns
+    predictors = [
+        "pclass",
+        "sex",
+        "age",
+        "sibsp",
+        "parch",
+    ]  # NOTE TO SELF - Test with more predictors in the future (and different type of response)
+    response = "survived"
+
+    # Determine if response is continuous or boolean
+    bool_check = response_type_check(df, response)
+
+    cont_list = []  # will store continuous predictors for random forest
+
+    # Loop through each predictor column
+    for predictor in predictors:
+        # Determine if the predictor is cat/cont
+        cat_check = cat_cont_check(df, predictor)
+
+        if cat_check is False:
+            cont_list.append(predictor)
+
+        # Automatically generate the necessary plot(s) to inspect it
+        if bool_check is True and cat_check is True:
+            bool_response_cat_predictor_plots(df, predictor, response)
+        elif bool_check is True and cat_check is False:
+            bool_response_cont_predictor_plots(df, predictor, response)
+        elif bool_check is False and cat_check is True:
+            cont_response_cat_predictor_plots(df, predictor, response)
+        else:
+            cont_response_cont_predictor_plots(df, predictor, response)
+
+        # Calculate the different ranking algos
+        # p-value & t-scole (continuous predictors only) along with it's plot
+        # Regression: Continuous response
+        # Logistic regression: Boolean response
+        if cat_check is False:
+            if bool_check is True:
+                logistic_regression(df, predictor, response)
+            else:
+                linear_regression(df, predictor, response)
+
+        # Difference with mean of response along with it's plot (weighted and unweighted)
+        diff_with_mean_of_resp(df, predictor, response, cat_check, bool_check)
+
+    # Random Forest Variable importance ranking (continuous predictors only)
+    random_forest(df, cont_list, response)
```

Comment:
since the file is testing with only one continuous predictor, the random forest importance came out to be 0 and 1, which is true, since age is this only predictor. I've tested with adding "fare" predictor to the function, it came out to be 0.593301 and 0.406699 on the importance scale, which is great. it ran great with more continuous predictors adding into the dataset. Great work here!